### DISCUSSION TITLE
Why methods used to register records for update are called registerDirty instead of registerUpdate?

### DISCUSSION THREAD

**Discussion by lDante93** [apex-enterprise-patterns/fflib-apex-common#431] [Q&A] [answered]
Hello.
I cannot find an answer anywhere to my simple question, so I am asking here :) 
Why methods used to register records for update are called registerDirty instead of registerUpdate?

Thanks

**Comment by wimvelzeboer** [comment, marked as answer]
Hi @lDante93 
The reason for this is that it is the standard terminology of the design pattern "Unit Of Work".
See also: https://martinfowler.com/eaaCatalog/unitOfWork.html

In the fflib_SObjectUnitOfWork is only the registerClean missing, I had a discussion about adding that a long while ago, but at the end it did not go through.

**Reply by lDante93** [reply]
Thanks for a response :) Well, do you know why that term was used instead of registerUpdate and what registerClean should do in opposite to registerDirty? Unfortunately I don't have Martin Fowler Patterns book to find it by myself :/

**Reply by wimvelzeboer** [reply]
>  Well, do you know why that term was used instead of registerUpdate

Because the design pattern is using the term registerDirty, all other programming languages/frameworks that I know use that same terminology.

 >  what registerClean should do in opposite to registerDirty
 
RegisterClean is a way to register a clean and unmodified record, so that when you register the same modified record twice (e.g. two different service methods using the same UnitOfWork), it will know which fields are actually changed, and is able to do an automatic merge.  Now, we need to provide a list of changed fields manually...
